pkg/controller/reconciler/deployment_reconciler_test.go

```diff
@@ -478,82 +496,10 @@ func makeDeployment(name types.NamespacedName) *appsv1.Deployment {
 	}
 }
 
-func waitForStateWaiting(t *testing.T, client client.Client, name types.NamespacedName) *deploymentAnnotations {
-	ctx := testcontext.New(t)
-
-	logger := t
-	var annotations deploymentAnnotations
-	require.EventuallyWithTf(t, func(t *assert.CollectT) {
-		logger.Logf("Fetching Deployment: %+v", name)
-		current := &appsv1.Deployment{}
-		err := client.Get(ctx, name, current)
-		require.NoError(t, err)
-
-		annotations, err = readAnnotations(current)
-		require.NoError(t, err)
-		assert.NotNil(t, annotations)
-		logger.Logf("Annotations.Status: %+v", annotations.Status)
-
-		if assert.NotNil(t, annotations.Status) && assert.Equal(t, deploymentPhraseWaiting, annotations.Status.Phrase) {
-			assert.Empty(t, annotations.Status.Operation)
-		}
-	}, deploymentTestWaitDuration, deploymentTestWaitInterval, "waiting for state to be Waiting")
-
-	return &annotations
-}
-
-func waitForStateUpdating(t *testing.T, client client.Client, name types.NamespacedName) *deploymentAnnotations {
-	ctx := testcontext.New(t)
-
-	logger := t
-	var annotations deploymentAnnotations
-	require.EventuallyWithTf(t, func(t *assert.CollectT) {
-		logger.Logf("Fetching Deployment: %+v", name)
-		current := &appsv1.Deployment{}
-		err := client.Get(ctx, name, current)
-		require.NoError(t, err)
-
-		annotations, err = readAnnotations(current)
-		require.NoError(t, err)
-		assert.NotNil(t, annotations)
-		logger.Logf("Annotations.Status: %+v", annotations.Status)
-
-		if assert.NotNil(t, annotations.Status) && assert.Equal(t, deploymentPhraseUpdating, annotations.Status.Phrase) {
-			assert.NotEmpty(t, annotations.Status.Operation)
-		}
-	}, deploymentTestWaitDuration, deploymentTestWaitInterval, "waiting for state to be Updating")
-
-	return &annotations
-}
-
-func waitForStateReady(t *testing.T, client client.Client, name types.NamespacedName) *deploymentAnnotations {
+func waitForState(t *testing.T, client client.Client, name types.NamespacedName, phrase deploymentPhrase) *deploymentAnnotations {
```

Comment:
Not sure if this is a good in terms of refactoring this code. What do you think @rynowak?